nbconvert/filters/strings.py

```diff
@@ -86,27 +88,34 @@ def _convert_header_id(header_contents):
     """
     return header_contents.replace(' ', '-')
 
-def add_anchor(html):
-    """Add an id and an anchor-link to an html header
+
+class AddAnchor(LoggingConfigurable):
     
-    For use on markdown headings
-    """
-    try:
-        h = ElementTree.fromstring(py3compat.cast_bytes_py2(html, encoding='utf-8'))
-    except Exception:
-        # failed to parse, just return it unmodified
-        return html
-    link = _convert_header_id(html2text(h))
-    h.set('id', link)
-    a = ElementTree.Element("a", {"class" : "anchor-link", "href" : "#" + link})
-    a.text = u'¶'
-    h.append(a)
-
-    # Known issue of Python3.x, ElementTree.tostring() returns a byte string
-    # instead of a text string.  See issue http://bugs.python.org/issue10942
-    # Workaround is to make sure the bytes are casted to a string.
-    return py3compat.decode(ElementTree.tostring(h), 'utf-8')
+    anchor_link_text = Unicode(u'¶',
+            help="The text used as the target for anchor links.").tag(config=True)
+    
+    def __init__(self, **kw):
+        super(AddAnchor, self).__init__(**kw)
 
+    def __call__(self, html, **kw):
+        
+        try:
+            h = ElementTree.fromstring(py3compat.cast_bytes_py2(html, encoding='utf-8'))
+        except Exception:
+            # failed to parse, just return it unmodified
+            return html
+
+        link = _convert_header_id(html2text(h))
+        h.set('id', link)
+        a = ElementTree.Element("a", {"class" : "anchor-link", "href" : "#" + link})
+        a.text = self.anchor_link_text
+        h.append(a)
+        # Known issue of Python3.x, ElementTree.tostring() returns a byte string
+        # instead of a text string.  See issue http://bugs.python.org/issue10942
+        # Workaround is to make sure the bytes are casted to a string.
+        return py3compat.decode(ElementTree.tostring(h), 'utf-8')
+
+add_anchor = AddAnchor
```

Comment:
are you imagining that this also would have a deprecation message?

```python
def add_anchor():
    warnings.warn("deprecation message")
    return AddAnchor()
```

and also, since elsewhere we create the object with configuration parameters, would we want to pass those in as optional arguments? 

```python
def add_anchor(local_config=""):
    warnings.warn("deprecation message")
    return AddAnchor(local_config=local_config)
```

In which case do we really want to say that this is deprecated since we could use it instead of calling AddAnchor directly in the IpythonRenderer call?